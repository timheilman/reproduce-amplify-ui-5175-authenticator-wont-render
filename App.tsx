import { Amplify } from "aws-amplify";
import { Button } from "react-native";
import { libraryOptions, resourcesConfig } from "./amplifyConfig";
import { Authenticator, useAuthenticator } from "@aws-amplify/ui-react-native";
Amplify.configure(resourcesConfig, libraryOptions);

function SignOutButton() {
  const { signOut } = useAuthenticator();
  return <Button title="Sign Out" onPress={signOut} />;
}

function App() {
  return (
    <Authenticator.Provider>
      <Authenticator>
        <SignOutButton />
      </Authenticator>
    </Authenticator.Provider>
  );
}
export default App;
