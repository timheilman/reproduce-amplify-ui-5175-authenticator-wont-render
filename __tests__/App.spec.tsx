import { render, screen } from "@testing-library/react-native";

import App from "../App";
import React from "react";

describe("<App />", () => {
  test("Authenticator renders", async () => {
    const { toJSON } = render(<App />);
    expect(await screen.findByText("Sign In")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
