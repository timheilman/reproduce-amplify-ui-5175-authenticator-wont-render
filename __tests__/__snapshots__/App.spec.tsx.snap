// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Authenticator renders 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      [
        {
          "alignContent": "center",
          "backgroundColor": "hsl(0, 0%, 100%)",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceVertical={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "flexGrow": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        undefined,
      ]
    }
    keyboardShouldPersistTaps="handled"
    style={
      [
        {},
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          {
            "marginHorizontal": 16,
          }
        }
      >
        <Text
          accessibilityRole="header"
          style={
            [
              {
                "color": "hsl(210, 50%, 10%)",
              },
              {
                "fontSize": 24,
                "fontWeight": "500",
                "lineHeight": 36,
              },
              {
                "marginVertical": 12,
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          Sign In
        </Text>
        <View
          style={
            {
              "paddingBottom": 8,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "marginBottom": 8,
                },
                {
                  "paddingHorizontal": 12,
                },
              ]
            }
            testID="amplify__text-field-container"
          >
            <Text
              accessibilityLabel="Email"
              accessibilityRole="text"
              style={
                [
                  {
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  {
                    "color": "hsl(210, 50%, 10%)",
                  },
                  [
                    {
                      "color": "hsl(210, 25%, 25%)",
                      "paddingVertical": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Email
            </Text>
            <View
              accessible={true}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "hsl(210, 10%, 58%)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 8,
                }
              }
              testID="amplify__text-field__input-container"
            >
              <TextInput
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                autoCapitalize="none"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Enter your Email"
                placeholderTextColor="hsl(210, 10%, 40%)"
                required={true}
                style={
                  [
                    {
                      "color": "hsl(210, 50%, 10%)",
                      "flexGrow": 1,
                      "fontSize": 16,
                      "paddingHorizontal": 4,
                      "paddingVertical": 12,
                    },
                    {},
                  ]
                }
                testID="authenticator__text-field__input-username"
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "marginBottom": 8,
                },
                [
                  {},
                  {
                    "paddingHorizontal": 12,
                  },
                ],
              ]
            }
            testID="amplify__text-field-container"
          >
            <Text
              accessibilityLabel="Password"
              accessibilityRole="text"
              style={
                [
                  {
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  {
                    "color": "hsl(210, 50%, 10%)",
                  },
                  [
                    {
                      "color": "hsl(210, 25%, 25%)",
                      "paddingVertical": 8,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Password
            </Text>
            <View
              accessible={true}
              style={
                {
                  "alignItems": "center",
                  "borderColor": "hsl(210, 10%, 58%)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 8,
                }
              }
              testID="amplify__text-field__input-container"
            >
              <TextInput
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                autoCapitalize="none"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Enter your Password"
                placeholderTextColor="hsl(210, 10%, 40%)"
                required={true}
                secureTextEntry={true}
                style={
                  [
                    {
                      "color": "hsl(210, 50%, 10%)",
                      "flexGrow": 1,
                      "fontSize": 16,
                      "paddingHorizontal": 4,
                      "paddingVertical": 12,
                    },
                    {},
                  ]
                }
                testID="authenticator__text-field__input-password"
              />
              <View
                accessibilityLabel="Show password"
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {},
                    undefined,
                    undefined,
                  ]
                }
              >
                <Image
                  accessibilityRole="image"
                  accessible={true}
                  source={
                    {
                      "testUri": "../../@aws-amplify/ui-react-native/lib/assets/icons/visibilityOff.png",
                    }
                  }
                  style={
                    [
                      {
                        "height": 16,
                        "resizeMode": "contain",
                        "tintColor": undefined,
                        "width": 16,
                      },
                      [
                        {
                          "tintColor": "hsl(210, 10%, 40%)",
                        },
                        undefined,
                      ],
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <TextInput
            accessibilityElementsHidden={true}
            pointerEvents="none"
            style={
              {
                "backgroundColor": "transparent",
                "height": 0.1,
                "width": 0.1,
              }
            }
          />
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "hsl(190, 95%, 30%)",
                "borderRadius": 4,
                "flexDirection": "row",
                "gap": 12,
                "justifyContent": "center",
                "opacity": 0.6,
                "paddingHorizontal": 16,
                "paddingVertical": 12,
              },
              undefined,
              {
                "margin": 12,
              },
            ]
          }
          testID="amplify__button"
        >
          <Text
            style={
              [
                {
                  "color": "hsl(0, 0%, 100%)",
                  "fontSize": 16,
                  "fontWeight": "700",
                  "textAlign": "center",
                },
                {},
              ]
            }
          >
            Sign in
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "gap": 12,
                  "justifyContent": "center",
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "marginVertical": 8,
                  "minWidth": "50%",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "hsl(190, 95%, 30%)",
                    "fontSize": 16,
                    "fontWeight": "700",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Forgot Password?
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "gap": 12,
                  "justifyContent": "center",
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                },
                undefined,
                {
                  "marginVertical": 8,
                  "minWidth": "50%",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "hsl(190, 95%, 30%)",
                    "fontSize": 16,
                    "fontWeight": "700",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Create Account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
